### PR TITLE
[doc] Add plot_parallel examples and fix rankdata use

### PR DESCRIPTION
**Pull request by ahartikainen**
## Description

Add more examples plots. (normal, rankdata and minmax)

Use rankdata with `"average"`.

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

## Plots

- Default
mpl_plot_parallel

- minmax
mpl_plot_parallel_minmax

- normal
mpl_plot_parallel_normal

- rank
mpl_plot_parallel_rank

### CODE REVIEW

File: arviz/plots/backends/bokeh/essplot.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def plot_ess(
 
             values = data[var_name].sel(**selection).values.flatten()
             mask = idata.sample_stats[rug_kind].values.flatten()
-            values = rankdata(values)[mask]
+            values = rankdata(values, method="average")[mask]
```
Comment (aloctavodia) [inline arviz/plots/backends/bokeh/essplot.py:91]: method="average" is the default, right? So this is making it explicit, right?

File: examples/matplotlib/mpl_plot_parallel_rank.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+"""
+import matplotlib.pyplot as plt
+
+import arviz as az
+
+az.style.use("arviz-darkgrid")
+
+data = az.load_arviz_data("centered_eight")
+ax = az.plot_parallel(data, var_names=["theta", "tau", "mu"], norm_method="rank")
+ax.set_xticklabels(ax.get_xticklabels(), rotation=70)
```
Comment (aloctavodia) [inline examples/matplotlib/mpl_plot_parallel_rank.py:15]: :+1:

File: arviz/plots/backends/bokeh/essplot.py (hunk 1 above)
Comment (ahartikainen) [reply arviz/plots/backends/bokeh/essplot.py:91]: yes, nothing really changes

**Comment by ahartikainen** [comment]
Also, see the change in pipelines

https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=3378&view=results

There is published section, where we can download the example figures. What other things should we upload from html pages?
The total build folder is 60+MB, but I does it matter if we upload the whole folder? This would help users/devs to see changes in docs without need to build it locally.

**Comment by ahartikainen** [comment]
Ok, I will set it to upload full build doc, then everything works when needed.

I'm not sure, but I don't think uploading 60mb per successful pipeline is too bad. Users don't need to download them if they don't need it. And pipelines automatically delete old ones. Any opinions/comments?

**Comment by ahartikainen** [comment]
This is ready now.

**Comment by ahartikainen** [comment]
Ok, the build docs size is 93MB.

What if we upload docs dynamically (e.g. user adds [viz] or [doc] to PR header?)

[plot]
[viz]
[doc]
[docs]

**Comment by ahartikainen** [comment]
Now docs are build for every commit, this is a good thing.

It will upload the build docs to artifacts if branch is `master` or some of these can be found in the commit title [plot], [viz], [doc], [docs]

This way user can access/test docs locally without building them (locally).

User needs to download the artifact from the three dots button in the right side of the artifact page

https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=3382&view=artifacts&type=publishedArtifacts

**Comment by OriolAbril** [comment]
This is very cool, thanks! Do you think you can add some lines about this to the developer guide? Or maybe the contributing one? Not really sure what should go into which document yet

**Review by OriolAbril — APPROVED** [review]

**Comment by ahartikainen** [comment]
Yes we can add that info. Another PR?

**Comment by OriolAbril** [comment]
> Yes we can add that info. Another PR?

whatever works best for you

**Comment by ahartikainen** [comment]
I will wait with the merge. (still waiting for pymc3 to fix pickling problem)

**Comment by canyon289** [comment]
Were good now

**Pull request merged**